<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 2 Preparing data sets | Introduction to ASURAT (version 0.0.0.9000)</title>
  <meta name="description" content="This is a developmental version of ASURAT." />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 2 Preparing data sets | Introduction to ASURAT (version 0.0.0.9000)" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This is a developmental version of ASURAT." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 2 Preparing data sets | Introduction to ASURAT (version 0.0.0.9000)" />
  
  <meta name="twitter:description" content="This is a developmental version of ASURAT." />
  

<meta name="author" content="Keita Iida" />


<meta name="date" content="2021-06-09" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="index.html"/>
<link rel="next" href="dataqc.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A Minimal Book Example</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Overview</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#license-and-description"><i class="fa fa-check"></i><b>1.1</b> License and description</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#computational-environment"><i class="fa fa-check"></i><b>1.2</b> Computational environment</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#installation"><i class="fa fa-check"></i><b>1.3</b> Installation</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="dataset.html"><a href="dataset.html"><i class="fa fa-check"></i><b>2</b> Preparing data sets</a><ul>
<li class="chapter" data-level="2.1" data-path="dataset.html"><a href="dataset.html#case-i-input-gene-expression-matrix-with-row-and-column-names"><i class="fa fa-check"></i><b>2.1</b> Case I: input gene expression matrix with row and column names</a></li>
<li class="chapter" data-level="2.2" data-path="dataset.html"><a href="dataset.html#case-ii-input-10x-data"><i class="fa fa-check"></i><b>2.2</b> Case II: input 10x data</a></li>
<li class="chapter" data-level="2.3" data-path="dataset.html"><a href="dataset.html#case-iii-input-10x-pbmc-data"><i class="fa fa-check"></i><b>2.3</b> Case III: input 10x PBMC data</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="dataqc.html"><a href="dataqc.html"><i class="fa fa-check"></i><b>3</b> Data quality control (QC)</a><ul>
<li class="chapter" data-level="3.1" data-path="dataqc.html"><a href="dataqc.html#quick-qc"><i class="fa fa-check"></i><b>3.1</b> Quick QC</a></li>
<li class="chapter" data-level="3.2" data-path="dataqc.html"><a href="dataqc.html#sample-qc"><i class="fa fa-check"></i><b>3.2</b> Sample QC</a></li>
<li class="chapter" data-level="3.3" data-path="dataqc.html"><a href="dataqc.html#variable-qc"><i class="fa fa-check"></i><b>3.3</b> Variable QC</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="normalization.html"><a href="normalization.html"><i class="fa fa-check"></i><b>4</b> Normalizing and centering data</a></li>
<li class="chapter" data-level="5" data-path="computcorr.html"><a href="computcorr.html"><i class="fa fa-check"></i><b>5</b> Computing correlations among genes</a></li>
<li class="chapter" data-level="6" data-path="markergenes.html"><a href="markergenes.html"><i class="fa fa-check"></i><b>6</b> Checking expression profiles of marker genes</a></li>
<li class="chapter" data-level="7" data-path="collectdb.html"><a href="collectdb.html"><i class="fa fa-check"></i><b>7</b> Collecting databases (optional)</a><ul>
<li class="chapter" data-level="7.1" data-path="collectdb.html"><a href="collectdb.html#disease-ontology-db"><i class="fa fa-check"></i><b>7.1</b> Disease Ontology DB</a></li>
<li class="chapter" data-level="7.2" data-path="collectdb.html"><a href="collectdb.html#cell-ontology-db"><i class="fa fa-check"></i><b>7.2</b> Cell Ontology DB</a></li>
<li class="chapter" data-level="7.3" data-path="collectdb.html"><a href="collectdb.html#gene-ontology-db"><i class="fa fa-check"></i><b>7.3</b> Gene Ontology DB</a></li>
<li class="chapter" data-level="7.4" data-path="collectdb.html"><a href="collectdb.html#kegg"><i class="fa fa-check"></i><b>7.4</b> KEGG</a></li>
<li class="chapter" data-level="7.5" data-path="collectdb.html"><a href="collectdb.html#reactome"><i class="fa fa-check"></i><b>7.5</b> Reactome</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="asurat-using-disease-ontology-database.html"><a href="asurat-using-disease-ontology-database.html"><i class="fa fa-check"></i><b>8</b> ASURAT using Disease Ontology database</a><ul>
<li class="chapter" data-level="8.1" data-path="asurat-using-disease-ontology-database.html"><a href="asurat-using-disease-ontology-database.html#creating-sign-by-sample-matrices"><i class="fa fa-check"></i><b>8.1</b> Creating sign-by-sample matrices</a></li>
<li class="chapter" data-level="8.2" data-path="asurat-using-disease-ontology-database.html"><a href="asurat-using-disease-ontology-database.html#dimensionality-reduction"><i class="fa fa-check"></i><b>8.2</b> Dimensionality reduction</a></li>
<li class="chapter" data-level="8.3" data-path="asurat-using-disease-ontology-database.html"><a href="asurat-using-disease-ontology-database.html#cell-clustering"><i class="fa fa-check"></i><b>8.3</b> Cell clustering</a></li>
<li class="chapter" data-level="8.4" data-path="asurat-using-disease-ontology-database.html"><a href="asurat-using-disease-ontology-database.html#visualization"><i class="fa fa-check"></i><b>8.4</b> Visualization</a></li>
<li class="chapter" data-level="8.5" data-path="asurat-using-disease-ontology-database.html"><a href="asurat-using-disease-ontology-database.html#significant-signs"><i class="fa fa-check"></i><b>8.5</b> Significant signs</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="asurat-using-cell-ontology-database-optional.html"><a href="asurat-using-cell-ontology-database-optional.html"><i class="fa fa-check"></i><b>9</b> ASURAT using Cell Ontology database (optional)</a><ul>
<li class="chapter" data-level="9.1" data-path="asurat-using-cell-ontology-database-optional.html"><a href="asurat-using-cell-ontology-database-optional.html#creating-sign-by-sample-matrices-1"><i class="fa fa-check"></i><b>9.1</b> Creating sign-by-sample matrices</a></li>
<li class="chapter" data-level="9.2" data-path="asurat-using-cell-ontology-database-optional.html"><a href="asurat-using-cell-ontology-database-optional.html#dimensionality-reduction-1"><i class="fa fa-check"></i><b>9.2</b> Dimensionality reduction</a></li>
<li class="chapter" data-level="9.3" data-path="asurat-using-cell-ontology-database-optional.html"><a href="asurat-using-cell-ontology-database-optional.html#cell-clustering-1"><i class="fa fa-check"></i><b>9.3</b> Cell clustering</a></li>
<li class="chapter" data-level="9.4" data-path="asurat-using-cell-ontology-database-optional.html"><a href="asurat-using-cell-ontology-database-optional.html#visualization-1"><i class="fa fa-check"></i><b>9.4</b> Visualization</a></li>
<li class="chapter" data-level="9.5" data-path="asurat-using-cell-ontology-database-optional.html"><a href="asurat-using-cell-ontology-database-optional.html#significant-signs-1"><i class="fa fa-check"></i><b>9.5</b> Significant signs</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="asurat-using-gene-ontology-database-optional.html"><a href="asurat-using-gene-ontology-database-optional.html"><i class="fa fa-check"></i><b>10</b> ASURAT using Gene Ontology database (optional)</a><ul>
<li class="chapter" data-level="10.1" data-path="asurat-using-gene-ontology-database-optional.html"><a href="asurat-using-gene-ontology-database-optional.html#creating-sign-by-sample-matrices-2"><i class="fa fa-check"></i><b>10.1</b> Creating sign-by-sample matrices</a></li>
<li class="chapter" data-level="10.2" data-path="asurat-using-gene-ontology-database-optional.html"><a href="asurat-using-gene-ontology-database-optional.html#dimensionality-reduction-2"><i class="fa fa-check"></i><b>10.2</b> Dimensionality reduction</a></li>
<li class="chapter" data-level="10.3" data-path="asurat-using-gene-ontology-database-optional.html"><a href="asurat-using-gene-ontology-database-optional.html#cell-clustering-2"><i class="fa fa-check"></i><b>10.3</b> Cell clustering</a></li>
<li class="chapter" data-level="10.4" data-path="asurat-using-gene-ontology-database-optional.html"><a href="asurat-using-gene-ontology-database-optional.html#visualization-2"><i class="fa fa-check"></i><b>10.4</b> Visualization</a></li>
<li class="chapter" data-level="10.5" data-path="asurat-using-gene-ontology-database-optional.html"><a href="asurat-using-gene-ontology-database-optional.html#significant-signs-2"><i class="fa fa-check"></i><b>10.5</b> Significant signs</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="asurat-using-kegg-optional.html"><a href="asurat-using-kegg-optional.html"><i class="fa fa-check"></i><b>11</b> ASURAT using KEGG (optional)</a><ul>
<li class="chapter" data-level="11.1" data-path="asurat-using-kegg-optional.html"><a href="asurat-using-kegg-optional.html#creating-sign-by-sample-matrices-3"><i class="fa fa-check"></i><b>11.1</b> Creating sign-by-sample matrices</a></li>
<li class="chapter" data-level="11.2" data-path="asurat-using-kegg-optional.html"><a href="asurat-using-kegg-optional.html#dimensionality-reduction-3"><i class="fa fa-check"></i><b>11.2</b> Dimensionality reduction</a></li>
<li class="chapter" data-level="11.3" data-path="asurat-using-kegg-optional.html"><a href="asurat-using-kegg-optional.html#cell-clustering-3"><i class="fa fa-check"></i><b>11.3</b> Cell clustering</a></li>
<li class="chapter" data-level="11.4" data-path="asurat-using-kegg-optional.html"><a href="asurat-using-kegg-optional.html#visualizing"><i class="fa fa-check"></i><b>11.4</b> Visualizing</a></li>
<li class="chapter" data-level="11.5" data-path="asurat-using-kegg-optional.html"><a href="asurat-using-kegg-optional.html#significant-signs-3"><i class="fa fa-check"></i><b>11.5</b> Significant signs</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="asurat-using-reactome-optional.html"><a href="asurat-using-reactome-optional.html"><i class="fa fa-check"></i><b>12</b> ASURAT using Reactome (optional)</a><ul>
<li class="chapter" data-level="12.1" data-path="asurat-using-reactome-optional.html"><a href="asurat-using-reactome-optional.html#creating-sign-by-sample-matrices-4"><i class="fa fa-check"></i><b>12.1</b> Creating sign-by-sample matrices</a></li>
<li class="chapter" data-level="12.2" data-path="asurat-using-reactome-optional.html"><a href="asurat-using-reactome-optional.html#dimensionality-reduction-4"><i class="fa fa-check"></i><b>12.2</b> Dimensionality reduction</a></li>
<li class="chapter" data-level="12.3" data-path="asurat-using-reactome-optional.html"><a href="asurat-using-reactome-optional.html#sample-clustering"><i class="fa fa-check"></i><b>12.3</b> Sample clustering</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="multiple-sign-analysis-by-concatenating-do-co-go-kegg-and-reactome.html"><a href="multiple-sign-analysis-by-concatenating-do-co-go-kegg-and-reactome.html"><i class="fa fa-check"></i><b>13</b> Multiple sign analysis by concatenating DO, CO, GO, KEGG, and Reactome</a></li>
<li class="chapter" data-level="14" data-path="appendix-a-comparing-performances-of-asurat-and-existing-methods.html"><a href="appendix-a-comparing-performances-of-asurat-and-existing-methods.html"><i class="fa fa-check"></i><b>14</b> Appendix A: comparing performances of ASURAT and existing methods</a><ul>
<li class="chapter" data-level="14.1" data-path="appendix-a-comparing-performances-of-asurat-and-existing-methods.html"><a href="appendix-a-comparing-performances-of-asurat-and-existing-methods.html#installation-1"><i class="fa fa-check"></i><b>14.1</b> Installation</a></li>
<li class="chapter" data-level="14.2" data-path="appendix-a-comparing-performances-of-asurat-and-existing-methods.html"><a href="appendix-a-comparing-performances-of-asurat-and-existing-methods.html#removing-the-outlier-cells-detected-by-asurat"><i class="fa fa-check"></i><b>14.2</b> Removing the outlier cells detected by ASURAT</a></li>
<li class="chapter" data-level="14.3" data-path="appendix-a-comparing-performances-of-asurat-and-existing-methods.html"><a href="appendix-a-comparing-performances-of-asurat-and-existing-methods.html#assessing-clustering-performances"><i class="fa fa-check"></i><b>14.3</b> Assessing clustering performances</a><ul>
<li class="chapter" data-level="14.3.1" data-path="appendix-a-comparing-performances-of-asurat-and-existing-methods.html"><a href="appendix-a-comparing-performances-of-asurat-and-existing-methods.html#seurat"><i class="fa fa-check"></i><b>14.3.1</b> Seurat</a></li>
<li class="chapter" data-level="14.3.2" data-path="appendix-a-comparing-performances-of-asurat-and-existing-methods.html"><a href="appendix-a-comparing-performances-of-asurat-and-existing-methods.html#monocle-3"><i class="fa fa-check"></i><b>14.3.2</b> Monocle 3</a></li>
<li class="chapter" data-level="14.3.3" data-path="appendix-a-comparing-performances-of-asurat-and-existing-methods.html"><a href="appendix-a-comparing-performances-of-asurat-and-existing-methods.html#sc3"><i class="fa fa-check"></i><b>14.3.3</b> SC3</a></li>
<li class="chapter" data-level="14.3.4" data-path="appendix-a-comparing-performances-of-asurat-and-existing-methods.html"><a href="appendix-a-comparing-performances-of-asurat-and-existing-methods.html#principal-component-analysis-pca"><i class="fa fa-check"></i><b>14.3.4</b> Principal component analysis (PCA)</a></li>
<li class="chapter" data-level="14.3.5" data-path="appendix-a-comparing-performances-of-asurat-and-existing-methods.html"><a href="appendix-a-comparing-performances-of-asurat-and-existing-methods.html#asurat"><i class="fa fa-check"></i><b>14.3.5</b> ASURAT</a></li>
<li class="chapter" data-level="14.3.6" data-path="appendix-a-comparing-performances-of-asurat-and-existing-methods.html"><a href="appendix-a-comparing-performances-of-asurat-and-existing-methods.html#comparing-clustering-validity-indices"><i class="fa fa-check"></i><b>14.3.6</b> Comparing clustering validity indices</a></li>
</ul></li>
<li class="chapter" data-level="14.4" data-path="appendix-a-comparing-performances-of-asurat-and-existing-methods.html"><a href="appendix-a-comparing-performances-of-asurat-and-existing-methods.html#cell-type-inference-in-pbmc_4000-and-pbmc_6000"><i class="fa fa-check"></i><b>14.4</b> Cell-type inference in <code>pbmc_4000</code> and <code>pbmc_6000</code></a><ul>
<li class="chapter" data-level="14.4.1" data-path="appendix-a-comparing-performances-of-asurat-and-existing-methods.html"><a href="appendix-a-comparing-performances-of-asurat-and-existing-methods.html#seurat-1"><i class="fa fa-check"></i><b>14.4.1</b> Seurat</a></li>
<li class="chapter" data-level="14.4.2" data-path="appendix-a-comparing-performances-of-asurat-and-existing-methods.html"><a href="appendix-a-comparing-performances-of-asurat-and-existing-methods.html#monocle-3-1"><i class="fa fa-check"></i><b>14.4.2</b> Monocle 3</a></li>
<li class="chapter" data-level="14.4.3" data-path="appendix-a-comparing-performances-of-asurat-and-existing-methods.html"><a href="appendix-a-comparing-performances-of-asurat-and-existing-methods.html#sc3-1"><i class="fa fa-check"></i><b>14.4.3</b> SC3</a></li>
<li class="chapter" data-level="14.4.4" data-path="appendix-a-comparing-performances-of-asurat-and-existing-methods.html"><a href="appendix-a-comparing-performances-of-asurat-and-existing-methods.html#asurat-1"><i class="fa fa-check"></i><b>14.4.4</b> ASURAT</a></li>
</ul></li>
<li class="chapter" data-level="14.5" data-path="appendix-a-comparing-performances-of-asurat-and-existing-methods.html"><a href="appendix-a-comparing-performances-of-asurat-and-existing-methods.html#identifying-cell-types-in-pbmc_4000-and-pbmc_6000"><i class="fa fa-check"></i><b>14.5</b> Identifying cell types in <code>pbmc_4000</code> and <code>pbmc_6000</code></a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="appendix-b-automatically-tuning-asurats-parameters.html"><a href="appendix-b-automatically-tuning-asurats-parameters.html"><i class="fa fa-check"></i><b>15</b> Appendix B: automatically tuning ASURATâ€™s parameters</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Introduction to ASURAT (version 0.0.0.9000)</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="dataset" class="section level1">
<h1><span class="header-section-number">Chapter 2</span> Preparing data sets</h1>
<div id="case-i-input-gene-expression-matrix-with-row-and-column-names" class="section level2">
<h2><span class="header-section-number">2.1</span> Case I: input gene expression matrix with row and column names</h2>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="dataset.html#cb1-1"></a><span class="kw">source</span>(<span class="st">&quot;R/function_general.R&quot;</span>)</span></code></pre></div>
<p>The following single-cell RNA sequencing (scRNA-seq) data sets were generated
from cancer tissue originated spheroids (CTOSs) obtained from cervix cancer
patients, which were then processed by Nx1-seq protocol.
The accession numbers are DRR165196 and DRR165197.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="dataset.html#cb2-1"></a>mat1 &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="dt">file =</span> <span class="st">&quot;data/2020_001_iida/day1_norm/day1_norm.rds&quot;</span>)</span>
<span id="cb2-2"><a href="dataset.html#cb2-2"></a>mat7 &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="dt">file =</span> <span class="st">&quot;data/2020_001_iida/day7_hypo/day7_hypo.rds&quot;</span>)</span></code></pre></div>
<p>The following function <code>make_asurat_obj()</code> creates a structured object.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="dataset.html#cb3-1"></a>day1_norm &lt;-<span class="st"> </span><span class="kw">make_asurat_obj</span>(<span class="dt">mat =</span> mat1, <span class="dt">obj_name =</span> <span class="st">&quot;day1_norm&quot;</span>)</span>
<span id="cb3-2"><a href="dataset.html#cb3-2"></a>day7_hypo &lt;-<span class="st"> </span><span class="kw">make_asurat_obj</span>(<span class="dt">mat =</span> mat7, <span class="dt">obj_name =</span> <span class="st">&quot;day7_hypo&quot;</span>)</span></code></pre></div>
<p>Convert gene symbols into Entrez IDs by using <code>org.Hs.eg.db</code> package.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="dataset.html#cb4-1"></a>dictionary &lt;-<span class="st"> </span>AnnotationDbi<span class="op">::</span><span class="kw">select</span>(org.Hs.eg.db,</span>
<span id="cb4-2"><a href="dataset.html#cb4-2"></a>                                    <span class="dt">key =</span> day1_norm[[<span class="st">&quot;variable&quot;</span>]][[<span class="st">&quot;symbol&quot;</span>]],</span>
<span id="cb4-3"><a href="dataset.html#cb4-3"></a>                                    <span class="dt">columns =</span> <span class="kw">c</span>(<span class="st">&quot;ENTREZID&quot;</span>), <span class="dt">keytype =</span> <span class="st">&quot;SYMBOL&quot;</span>)</span>
<span id="cb4-4"><a href="dataset.html#cb4-4"></a>dictionary &lt;-<span class="st"> </span>dictionary[<span class="op">!</span><span class="kw">duplicated</span>(dictionary<span class="op">$</span>SYMBOL), ]</span>
<span id="cb4-5"><a href="dataset.html#cb4-5"></a><span class="kw">names</span>(dictionary) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;symbol&quot;</span>, <span class="st">&quot;entrez&quot;</span>)</span>
<span id="cb4-6"><a href="dataset.html#cb4-6"></a>day1_norm[[<span class="st">&quot;variable&quot;</span>]] &lt;-<span class="st"> </span>dictionary</span>
<span id="cb4-7"><a href="dataset.html#cb4-7"></a></span>
<span id="cb4-8"><a href="dataset.html#cb4-8"></a>dictionary &lt;-<span class="st"> </span>AnnotationDbi<span class="op">::</span><span class="kw">select</span>(org.Hs.eg.db,</span>
<span id="cb4-9"><a href="dataset.html#cb4-9"></a>                                    <span class="dt">key =</span> day7_hypo[[<span class="st">&quot;variable&quot;</span>]][[<span class="st">&quot;symbol&quot;</span>]],</span>
<span id="cb4-10"><a href="dataset.html#cb4-10"></a>                                    <span class="dt">columns =</span> <span class="kw">c</span>(<span class="st">&quot;ENTREZID&quot;</span>), <span class="dt">keytype =</span> <span class="st">&quot;SYMBOL&quot;</span>)</span>
<span id="cb4-11"><a href="dataset.html#cb4-11"></a>dictionary &lt;-<span class="st"> </span>dictionary[<span class="op">!</span><span class="kw">duplicated</span>(dictionary<span class="op">$</span>SYMBOL), ]</span>
<span id="cb4-12"><a href="dataset.html#cb4-12"></a><span class="kw">names</span>(dictionary) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;symbol&quot;</span>, <span class="st">&quot;entrez&quot;</span>)</span>
<span id="cb4-13"><a href="dataset.html#cb4-13"></a>day7_hypo[[<span class="st">&quot;variable&quot;</span>]] &lt;-<span class="st"> </span>dictionary</span></code></pre></div>
<p>Save the objects.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="dataset.html#cb5-1"></a><span class="kw">saveRDS</span>(day1_norm, <span class="dt">file =</span> <span class="st">&quot;backup/01_001_day1_norm_raw.rds&quot;</span>)</span>
<span id="cb5-2"><a href="dataset.html#cb5-2"></a><span class="kw">saveRDS</span>(day7_hypo, <span class="dt">file =</span> <span class="st">&quot;backup/02_001_day7_hypo_raw.rds&quot;</span>)</span></code></pre></div>
</div>
<div id="case-ii-input-10x-data" class="section level2">
<h2><span class="header-section-number">2.2</span> Case II: input 10x data</h2>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="dataset.html#cb6-1"></a><span class="kw">source</span>(<span class="st">&quot;R/function_general.R&quot;</span>)</span></code></pre></div>
<p>The data were obtained from NCBI repository with accession number <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE138474">GSE138474</a>,
from which the following SRA files were downloaded:</p>
<ul>
<li>SRR10211594 (for MDA-SC68 with vehicle treatment; sample accessions numbers
GSM4104165),</li>
<li>SRR10211593 (for MDA-SC68 with cisplatin treatment; sample accessions numbers
GSM4104164).</li>
</ul>
<p>The raw count matrices were dumped by 10x cellranger with a reference
genome GRCh38-3.0.0.
The following function <code>read_10xdata</code> processes the dumped files into a raw
count matrix.
In this function, <code>make.unique()</code> in <code>base</code> package is applied for naming gene
symbols, which appends a sequential number with a period delimiter for every
repeat name encountered.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="dataset.html#cb7-1"></a>read_10xdata &lt;-<span class="st"> </span><span class="cf">function</span>(path_dir){</span>
<span id="cb7-2"><a href="dataset.html#cb7-2"></a>  barcode.path &lt;-<span class="st"> </span><span class="kw">paste0</span>(path_dir, <span class="st">&quot;barcodes.tsv.gz&quot;</span>)</span>
<span id="cb7-3"><a href="dataset.html#cb7-3"></a>  feature.path &lt;-<span class="st"> </span><span class="kw">paste0</span>(path_dir, <span class="st">&quot;features.tsv.gz&quot;</span>)</span>
<span id="cb7-4"><a href="dataset.html#cb7-4"></a>  matrix.path  &lt;-<span class="st"> </span><span class="kw">paste0</span>(path_dir, <span class="st">&quot;matrix.mtx.gz&quot;</span>)</span>
<span id="cb7-5"><a href="dataset.html#cb7-5"></a>  mat &lt;-<span class="st"> </span><span class="kw">as.matrix</span>(<span class="kw">readMM</span>(<span class="dt">file =</span> matrix.path))</span>
<span id="cb7-6"><a href="dataset.html#cb7-6"></a>  genes &lt;-<span class="st"> </span><span class="kw">read.delim</span>(feature.path, <span class="dt">header =</span> <span class="ot">FALSE</span>, <span class="dt">stringsAsFactors =</span> <span class="ot">FALSE</span>)</span>
<span id="cb7-7"><a href="dataset.html#cb7-7"></a>  barcodes &lt;-<span class="st"> </span><span class="kw">read.delim</span>(barcode.path, <span class="dt">header =</span> <span class="ot">FALSE</span>, <span class="dt">stringsAsFactors =</span> <span class="ot">FALSE</span>)</span>
<span id="cb7-8"><a href="dataset.html#cb7-8"></a>  <span class="kw">rownames</span>(mat) &lt;-<span class="st"> </span><span class="kw">make.unique</span>(<span class="kw">as.character</span>(genes<span class="op">$</span>V2))</span>
<span id="cb7-9"><a href="dataset.html#cb7-9"></a>  <span class="kw">colnames</span>(mat) &lt;-<span class="st"> </span>barcodes<span class="op">$</span>V1</span>
<span id="cb7-10"><a href="dataset.html#cb7-10"></a></span>
<span id="cb7-11"><a href="dataset.html#cb7-11"></a>  <span class="kw">return</span>(mat)</span>
<span id="cb7-12"><a href="dataset.html#cb7-12"></a>}</span>
<span id="cb7-13"><a href="dataset.html#cb7-13"></a>path_dir1 &lt;-<span class="st"> &quot;data/2020_001_stewart/sc68_vehi/SRR10211594_count/&quot;</span></span>
<span id="cb7-14"><a href="dataset.html#cb7-14"></a>path_dir1 &lt;-<span class="st"> </span><span class="kw">paste</span>(path_dir1, <span class="st">&quot;filtered_feature_bc_matrix/&quot;</span>, <span class="dt">sep =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb7-15"><a href="dataset.html#cb7-15"></a>path_dir2 &lt;-<span class="st"> &quot;data/2020_001_stewart/sc68_cisp/SRR10211593_count/&quot;</span></span>
<span id="cb7-16"><a href="dataset.html#cb7-16"></a>path_dir2 &lt;-<span class="st"> </span><span class="kw">paste</span>(path_dir2, <span class="st">&quot;filtered_feature_bc_matrix/&quot;</span>, <span class="dt">sep =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb7-17"><a href="dataset.html#cb7-17"></a>sc68_vehi &lt;-<span class="st"> </span><span class="kw">make_asurat_obj</span>(<span class="dt">mat =</span> <span class="kw">read_10xdata</span>(<span class="dt">path_dir =</span> path_dir1),</span>
<span id="cb7-18"><a href="dataset.html#cb7-18"></a>                             <span class="dt">obj_name =</span> <span class="st">&quot;sc68_vehi&quot;</span>)</span>
<span id="cb7-19"><a href="dataset.html#cb7-19"></a>sc68_cisp &lt;-<span class="st"> </span><span class="kw">make_asurat_obj</span>(<span class="dt">mat =</span> <span class="kw">read_10xdata</span>(<span class="dt">path_dir =</span> path_dir2),</span>
<span id="cb7-20"><a href="dataset.html#cb7-20"></a>                             <span class="dt">obj_name =</span> <span class="st">&quot;sc68_cisp&quot;</span>)</span></code></pre></div>
<p>Convert gene symbols into Entrez IDs by using <code>org.Hs.eg.db</code> package.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="dataset.html#cb8-1"></a>dictionary &lt;-<span class="st"> </span>AnnotationDbi<span class="op">::</span><span class="kw">select</span>(org.Hs.eg.db,</span>
<span id="cb8-2"><a href="dataset.html#cb8-2"></a>                                    <span class="dt">key =</span> sc68_vehi[[<span class="st">&quot;variable&quot;</span>]][[<span class="st">&quot;symbol&quot;</span>]],</span>
<span id="cb8-3"><a href="dataset.html#cb8-3"></a>                                    <span class="dt">columns =</span> <span class="kw">c</span>(<span class="st">&quot;ENTREZID&quot;</span>), <span class="dt">keytype =</span> <span class="st">&quot;SYMBOL&quot;</span>)</span>
<span id="cb8-4"><a href="dataset.html#cb8-4"></a>dictionary &lt;-<span class="st"> </span>dictionary[<span class="op">!</span><span class="kw">duplicated</span>(dictionary<span class="op">$</span>SYMBOL), ]</span>
<span id="cb8-5"><a href="dataset.html#cb8-5"></a><span class="kw">names</span>(dictionary) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;symbol&quot;</span>, <span class="st">&quot;entrez&quot;</span>)</span>
<span id="cb8-6"><a href="dataset.html#cb8-6"></a>sc68_vehi[[<span class="st">&quot;variable&quot;</span>]] &lt;-<span class="st"> </span>dictionary</span>
<span id="cb8-7"><a href="dataset.html#cb8-7"></a></span>
<span id="cb8-8"><a href="dataset.html#cb8-8"></a>dictionary &lt;-<span class="st"> </span>AnnotationDbi<span class="op">::</span><span class="kw">select</span>(org.Hs.eg.db,</span>
<span id="cb8-9"><a href="dataset.html#cb8-9"></a>                                    <span class="dt">key =</span> sc68_cisp[[<span class="st">&quot;variable&quot;</span>]][[<span class="st">&quot;symbol&quot;</span>]],</span>
<span id="cb8-10"><a href="dataset.html#cb8-10"></a>                                    <span class="dt">columns =</span> <span class="kw">c</span>(<span class="st">&quot;ENTREZID&quot;</span>), <span class="dt">keytype =</span> <span class="st">&quot;SYMBOL&quot;</span>)</span>
<span id="cb8-11"><a href="dataset.html#cb8-11"></a>dictionary &lt;-<span class="st"> </span>dictionary[<span class="op">!</span><span class="kw">duplicated</span>(dictionary<span class="op">$</span>SYMBOL), ]</span>
<span id="cb8-12"><a href="dataset.html#cb8-12"></a><span class="kw">names</span>(dictionary) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;symbol&quot;</span>, <span class="st">&quot;entrez&quot;</span>)</span>
<span id="cb8-13"><a href="dataset.html#cb8-13"></a>sc68_cisp[[<span class="st">&quot;variable&quot;</span>]] &lt;-<span class="st"> </span>dictionary</span></code></pre></div>
<p>Save the objects.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="dataset.html#cb9-1"></a><span class="kw">saveRDS</span>(sc68_vehi, <span class="dt">file =</span> <span class="st">&quot;backup/03_001_sc68_vehi_raw.rds&quot;</span>)</span>
<span id="cb9-2"><a href="dataset.html#cb9-2"></a><span class="kw">saveRDS</span>(sc68_cisp, <span class="dt">file =</span> <span class="st">&quot;backup/04_001_sc68_cisp_raw.rds&quot;</span>)</span></code></pre></div>
</div>
<div id="case-iii-input-10x-pbmc-data" class="section level2">
<h2><span class="header-section-number">2.3</span> Case III: input 10x PBMC data</h2>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="dataset.html#cb10-1"></a><span class="kw">source</span>(<span class="st">&quot;R/function_general.R&quot;</span>)</span></code></pre></div>
<p>The data (peripheral blood mononuclear cells from a healthy donor) were
obtained from 10x Genomics repository (PBMC 4k and 6k).</p>
<p>Load the data.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="dataset.html#cb11-1"></a>read_10xdata_v2 &lt;-<span class="st"> </span><span class="cf">function</span>(path_dir){</span>
<span id="cb11-2"><a href="dataset.html#cb11-2"></a>  <span class="kw">library</span>(Seurat)</span>
<span id="cb11-3"><a href="dataset.html#cb11-3"></a>  obj &lt;-<span class="st"> </span><span class="kw">Read10X</span>(<span class="dt">data.dir =</span> path_dir, <span class="dt">gene.column =</span> <span class="dv">2</span>, <span class="dt">unique.features =</span> <span class="ot">TRUE</span>,</span>
<span id="cb11-4"><a href="dataset.html#cb11-4"></a>                 <span class="dt">strip.suffix =</span> <span class="ot">FALSE</span>)</span>
<span id="cb11-5"><a href="dataset.html#cb11-5"></a>  obj &lt;-<span class="st"> </span><span class="kw">CreateSeuratObject</span>(<span class="dt">counts =</span> obj)</span>
<span id="cb11-6"><a href="dataset.html#cb11-6"></a>  mat &lt;-<span class="st"> </span><span class="kw">as.matrix</span>(obj<span class="op">@</span>assays[[<span class="st">&quot;RNA&quot;</span>]]<span class="op">@</span>counts)</span>
<span id="cb11-7"><a href="dataset.html#cb11-7"></a>  <span class="kw">return</span>(mat)</span>
<span id="cb11-8"><a href="dataset.html#cb11-8"></a>}</span>
<span id="cb11-9"><a href="dataset.html#cb11-9"></a>path_dir1 &lt;-<span class="st"> &quot;data/2020_001_10xgenomics/pbmc_4000/&quot;</span></span>
<span id="cb11-10"><a href="dataset.html#cb11-10"></a>path_dir1 &lt;-<span class="st"> </span><span class="kw">paste</span>(path_dir1, <span class="st">&quot;filtered_gene_bc_matrices/GRCh38/&quot;</span>, <span class="dt">sep =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb11-11"><a href="dataset.html#cb11-11"></a>path_dir2 &lt;-<span class="st"> &quot;data/2020_001_10xgenomics/pbmc_6000/&quot;</span></span>
<span id="cb11-12"><a href="dataset.html#cb11-12"></a>path_dir2 &lt;-<span class="st"> </span><span class="kw">paste</span>(path_dir2, <span class="st">&quot;filtered_matrices_mex/hg19/&quot;</span>, <span class="dt">sep =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb11-13"><a href="dataset.html#cb11-13"></a>pbmc_<span class="dv">4000</span> &lt;-<span class="st"> </span><span class="kw">make_asurat_obj</span>(<span class="dt">mat =</span> <span class="kw">read_10xdata_v2</span>(<span class="dt">path_dir =</span> path_dir1),</span>
<span id="cb11-14"><a href="dataset.html#cb11-14"></a>                             <span class="dt">obj_name =</span> <span class="st">&quot;pbmc_4000&quot;</span>)</span>
<span id="cb11-15"><a href="dataset.html#cb11-15"></a>pbmc_<span class="dv">6000</span> &lt;-<span class="st"> </span><span class="kw">make_asurat_obj</span>(<span class="dt">mat =</span> <span class="kw">read_10xdata_v2</span>(<span class="dt">path_dir =</span> path_dir2),</span>
<span id="cb11-16"><a href="dataset.html#cb11-16"></a>                             <span class="dt">obj_name =</span> <span class="st">&quot;pbmc_6000&quot;</span>)</span></code></pre></div>
<p>Convert gene symbols into Entrez IDs by using <code>org.Hs.eg.db</code> package.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="dataset.html#cb12-1"></a>dictionary &lt;-<span class="st"> </span>AnnotationDbi<span class="op">::</span><span class="kw">select</span>(org.Hs.eg.db,</span>
<span id="cb12-2"><a href="dataset.html#cb12-2"></a>                                    <span class="dt">key =</span> pbmc_<span class="dv">4000</span>[[<span class="st">&quot;variable&quot;</span>]][[<span class="st">&quot;symbol&quot;</span>]],</span>
<span id="cb12-3"><a href="dataset.html#cb12-3"></a>                                    <span class="dt">columns =</span> <span class="kw">c</span>(<span class="st">&quot;ENTREZID&quot;</span>), <span class="dt">keytype =</span> <span class="st">&quot;SYMBOL&quot;</span>)</span>
<span id="cb12-4"><a href="dataset.html#cb12-4"></a>dictionary &lt;-<span class="st"> </span>dictionary[<span class="op">!</span><span class="kw">duplicated</span>(dictionary<span class="op">$</span>SYMBOL), ]</span>
<span id="cb12-5"><a href="dataset.html#cb12-5"></a><span class="kw">names</span>(dictionary) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;symbol&quot;</span>, <span class="st">&quot;entrez&quot;</span>)</span>
<span id="cb12-6"><a href="dataset.html#cb12-6"></a>pbmc_<span class="dv">4000</span>[[<span class="st">&quot;variable&quot;</span>]] &lt;-<span class="st"> </span>dictionary</span>
<span id="cb12-7"><a href="dataset.html#cb12-7"></a></span>
<span id="cb12-8"><a href="dataset.html#cb12-8"></a>dictionary &lt;-<span class="st"> </span>AnnotationDbi<span class="op">::</span><span class="kw">select</span>(org.Hs.eg.db,</span>
<span id="cb12-9"><a href="dataset.html#cb12-9"></a>                                    <span class="dt">key =</span> pbmc_<span class="dv">6000</span>[[<span class="st">&quot;variable&quot;</span>]][[<span class="st">&quot;symbol&quot;</span>]],</span>
<span id="cb12-10"><a href="dataset.html#cb12-10"></a>                                    <span class="dt">columns =</span> <span class="kw">c</span>(<span class="st">&quot;ENTREZID&quot;</span>), <span class="dt">keytype =</span> <span class="st">&quot;SYMBOL&quot;</span>)</span>
<span id="cb12-11"><a href="dataset.html#cb12-11"></a>dictionary &lt;-<span class="st"> </span>dictionary[<span class="op">!</span><span class="kw">duplicated</span>(dictionary<span class="op">$</span>SYMBOL), ]</span>
<span id="cb12-12"><a href="dataset.html#cb12-12"></a><span class="kw">names</span>(dictionary) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;symbol&quot;</span>, <span class="st">&quot;entrez&quot;</span>)</span>
<span id="cb12-13"><a href="dataset.html#cb12-13"></a>pbmc_<span class="dv">6000</span>[[<span class="st">&quot;variable&quot;</span>]] &lt;-<span class="st"> </span>dictionary</span></code></pre></div>
<p>Save the objects.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="dataset.html#cb13-1"></a><span class="kw">saveRDS</span>(pbmc_<span class="dv">4000</span>, <span class="dt">file =</span> <span class="st">&quot;backup/05_001_pbmc_4000_raw.rds&quot;</span>)</span>
<span id="cb13-2"><a href="dataset.html#cb13-2"></a><span class="kw">saveRDS</span>(pbmc_<span class="dv">6000</span>, <span class="dt">file =</span> <span class="st">&quot;backup/06_001_pbmc_6000_raw.rds&quot;</span>)</span></code></pre></div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="index.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="dataqc.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["Introduction to ASURAT (ver. 0.0.0.9000).pdf", "Introduction to ASURAT (ver. 0.0.0.9000).epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
