<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 4 Normalizing and centering data | Introduction to ASURAT (version 0.0.0.9000)</title>
  <meta name="description" content="This is a developmental version of ASURAT." />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 4 Normalizing and centering data | Introduction to ASURAT (version 0.0.0.9000)" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This is a developmental version of ASURAT." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 4 Normalizing and centering data | Introduction to ASURAT (version 0.0.0.9000)" />
  
  <meta name="twitter:description" content="This is a developmental version of ASURAT." />
  

<meta name="author" content="Keita Iida" />


<meta name="date" content="2021-06-09" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="dataqc.html"/>
<link rel="next" href="computcorr.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A Minimal Book Example</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Overview</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#license-and-description"><i class="fa fa-check"></i><b>1.1</b> License and description</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#computational-environment"><i class="fa fa-check"></i><b>1.2</b> Computational environment</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#installation"><i class="fa fa-check"></i><b>1.3</b> Installation</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="dataset.html"><a href="dataset.html"><i class="fa fa-check"></i><b>2</b> Preparing data sets</a><ul>
<li class="chapter" data-level="2.1" data-path="dataset.html"><a href="dataset.html#case-i-input-gene-expression-matrix-with-row-and-column-names"><i class="fa fa-check"></i><b>2.1</b> Case I: input gene expression matrix with row and column names</a></li>
<li class="chapter" data-level="2.2" data-path="dataset.html"><a href="dataset.html#case-ii-input-10x-data"><i class="fa fa-check"></i><b>2.2</b> Case II: input 10x data</a></li>
<li class="chapter" data-level="2.3" data-path="dataset.html"><a href="dataset.html#case-iii-input-10x-pbmc-data"><i class="fa fa-check"></i><b>2.3</b> Case III: input 10x PBMC data</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="dataqc.html"><a href="dataqc.html"><i class="fa fa-check"></i><b>3</b> Data quality control (QC)</a><ul>
<li class="chapter" data-level="3.1" data-path="dataqc.html"><a href="dataqc.html#quick-qc"><i class="fa fa-check"></i><b>3.1</b> Quick QC</a></li>
<li class="chapter" data-level="3.2" data-path="dataqc.html"><a href="dataqc.html#sample-qc"><i class="fa fa-check"></i><b>3.2</b> Sample QC</a></li>
<li class="chapter" data-level="3.3" data-path="dataqc.html"><a href="dataqc.html#variable-qc"><i class="fa fa-check"></i><b>3.3</b> Variable QC</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="normalization.html"><a href="normalization.html"><i class="fa fa-check"></i><b>4</b> Normalizing and centering data</a></li>
<li class="chapter" data-level="5" data-path="computcorr.html"><a href="computcorr.html"><i class="fa fa-check"></i><b>5</b> Computing correlations among genes</a></li>
<li class="chapter" data-level="6" data-path="markergenes.html"><a href="markergenes.html"><i class="fa fa-check"></i><b>6</b> Checking expression profiles of marker genes</a></li>
<li class="chapter" data-level="7" data-path="collectdb.html"><a href="collectdb.html"><i class="fa fa-check"></i><b>7</b> Collecting databases (optional)</a><ul>
<li class="chapter" data-level="7.1" data-path="collectdb.html"><a href="collectdb.html#disease-ontology-db"><i class="fa fa-check"></i><b>7.1</b> Disease Ontology DB</a></li>
<li class="chapter" data-level="7.2" data-path="collectdb.html"><a href="collectdb.html#cell-ontology-db"><i class="fa fa-check"></i><b>7.2</b> Cell Ontology DB</a></li>
<li class="chapter" data-level="7.3" data-path="collectdb.html"><a href="collectdb.html#gene-ontology-db"><i class="fa fa-check"></i><b>7.3</b> Gene Ontology DB</a></li>
<li class="chapter" data-level="7.4" data-path="collectdb.html"><a href="collectdb.html#kegg"><i class="fa fa-check"></i><b>7.4</b> KEGG</a></li>
<li class="chapter" data-level="7.5" data-path="collectdb.html"><a href="collectdb.html#reactome"><i class="fa fa-check"></i><b>7.5</b> Reactome</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="asurat-using-disease-ontology-database.html"><a href="asurat-using-disease-ontology-database.html"><i class="fa fa-check"></i><b>8</b> ASURAT using Disease Ontology database</a><ul>
<li class="chapter" data-level="8.1" data-path="asurat-using-disease-ontology-database.html"><a href="asurat-using-disease-ontology-database.html#creating-sign-by-sample-matrices"><i class="fa fa-check"></i><b>8.1</b> Creating sign-by-sample matrices</a></li>
<li class="chapter" data-level="8.2" data-path="asurat-using-disease-ontology-database.html"><a href="asurat-using-disease-ontology-database.html#dimensionality-reduction"><i class="fa fa-check"></i><b>8.2</b> Dimensionality reduction</a></li>
<li class="chapter" data-level="8.3" data-path="asurat-using-disease-ontology-database.html"><a href="asurat-using-disease-ontology-database.html#cell-clustering"><i class="fa fa-check"></i><b>8.3</b> Cell clustering</a></li>
<li class="chapter" data-level="8.4" data-path="asurat-using-disease-ontology-database.html"><a href="asurat-using-disease-ontology-database.html#visualization"><i class="fa fa-check"></i><b>8.4</b> Visualization</a></li>
<li class="chapter" data-level="8.5" data-path="asurat-using-disease-ontology-database.html"><a href="asurat-using-disease-ontology-database.html#significant-signs"><i class="fa fa-check"></i><b>8.5</b> Significant signs</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="asurat-using-cell-ontology-database-optional.html"><a href="asurat-using-cell-ontology-database-optional.html"><i class="fa fa-check"></i><b>9</b> ASURAT using Cell Ontology database (optional)</a><ul>
<li class="chapter" data-level="9.1" data-path="asurat-using-cell-ontology-database-optional.html"><a href="asurat-using-cell-ontology-database-optional.html#creating-sign-by-sample-matrices-1"><i class="fa fa-check"></i><b>9.1</b> Creating sign-by-sample matrices</a></li>
<li class="chapter" data-level="9.2" data-path="asurat-using-cell-ontology-database-optional.html"><a href="asurat-using-cell-ontology-database-optional.html#dimensionality-reduction-1"><i class="fa fa-check"></i><b>9.2</b> Dimensionality reduction</a></li>
<li class="chapter" data-level="9.3" data-path="asurat-using-cell-ontology-database-optional.html"><a href="asurat-using-cell-ontology-database-optional.html#cell-clustering-1"><i class="fa fa-check"></i><b>9.3</b> Cell clustering</a></li>
<li class="chapter" data-level="9.4" data-path="asurat-using-cell-ontology-database-optional.html"><a href="asurat-using-cell-ontology-database-optional.html#visualization-1"><i class="fa fa-check"></i><b>9.4</b> Visualization</a></li>
<li class="chapter" data-level="9.5" data-path="asurat-using-cell-ontology-database-optional.html"><a href="asurat-using-cell-ontology-database-optional.html#significant-signs-1"><i class="fa fa-check"></i><b>9.5</b> Significant signs</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="asurat-using-gene-ontology-database-optional.html"><a href="asurat-using-gene-ontology-database-optional.html"><i class="fa fa-check"></i><b>10</b> ASURAT using Gene Ontology database (optional)</a><ul>
<li class="chapter" data-level="10.1" data-path="asurat-using-gene-ontology-database-optional.html"><a href="asurat-using-gene-ontology-database-optional.html#creating-sign-by-sample-matrices-2"><i class="fa fa-check"></i><b>10.1</b> Creating sign-by-sample matrices</a></li>
<li class="chapter" data-level="10.2" data-path="asurat-using-gene-ontology-database-optional.html"><a href="asurat-using-gene-ontology-database-optional.html#dimensionality-reduction-2"><i class="fa fa-check"></i><b>10.2</b> Dimensionality reduction</a></li>
<li class="chapter" data-level="10.3" data-path="asurat-using-gene-ontology-database-optional.html"><a href="asurat-using-gene-ontology-database-optional.html#cell-clustering-2"><i class="fa fa-check"></i><b>10.3</b> Cell clustering</a></li>
<li class="chapter" data-level="10.4" data-path="asurat-using-gene-ontology-database-optional.html"><a href="asurat-using-gene-ontology-database-optional.html#visualization-2"><i class="fa fa-check"></i><b>10.4</b> Visualization</a></li>
<li class="chapter" data-level="10.5" data-path="asurat-using-gene-ontology-database-optional.html"><a href="asurat-using-gene-ontology-database-optional.html#significant-signs-2"><i class="fa fa-check"></i><b>10.5</b> Significant signs</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="asurat-using-kegg-optional.html"><a href="asurat-using-kegg-optional.html"><i class="fa fa-check"></i><b>11</b> ASURAT using KEGG (optional)</a><ul>
<li class="chapter" data-level="11.1" data-path="asurat-using-kegg-optional.html"><a href="asurat-using-kegg-optional.html#creating-sign-by-sample-matrices-3"><i class="fa fa-check"></i><b>11.1</b> Creating sign-by-sample matrices</a></li>
<li class="chapter" data-level="11.2" data-path="asurat-using-kegg-optional.html"><a href="asurat-using-kegg-optional.html#dimensionality-reduction-3"><i class="fa fa-check"></i><b>11.2</b> Dimensionality reduction</a></li>
<li class="chapter" data-level="11.3" data-path="asurat-using-kegg-optional.html"><a href="asurat-using-kegg-optional.html#cell-clustering-3"><i class="fa fa-check"></i><b>11.3</b> Cell clustering</a></li>
<li class="chapter" data-level="11.4" data-path="asurat-using-kegg-optional.html"><a href="asurat-using-kegg-optional.html#visualizing"><i class="fa fa-check"></i><b>11.4</b> Visualizing</a></li>
<li class="chapter" data-level="11.5" data-path="asurat-using-kegg-optional.html"><a href="asurat-using-kegg-optional.html#significant-signs-3"><i class="fa fa-check"></i><b>11.5</b> Significant signs</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="asurat-using-reactome-optional.html"><a href="asurat-using-reactome-optional.html"><i class="fa fa-check"></i><b>12</b> ASURAT using Reactome (optional)</a><ul>
<li class="chapter" data-level="12.1" data-path="asurat-using-reactome-optional.html"><a href="asurat-using-reactome-optional.html#creating-sign-by-sample-matrices-4"><i class="fa fa-check"></i><b>12.1</b> Creating sign-by-sample matrices</a></li>
<li class="chapter" data-level="12.2" data-path="asurat-using-reactome-optional.html"><a href="asurat-using-reactome-optional.html#dimensionality-reduction-4"><i class="fa fa-check"></i><b>12.2</b> Dimensionality reduction</a></li>
<li class="chapter" data-level="12.3" data-path="asurat-using-reactome-optional.html"><a href="asurat-using-reactome-optional.html#sample-clustering"><i class="fa fa-check"></i><b>12.3</b> Sample clustering</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="multiple-sign-analysis-by-concatenating-do-co-go-kegg-and-reactome.html"><a href="multiple-sign-analysis-by-concatenating-do-co-go-kegg-and-reactome.html"><i class="fa fa-check"></i><b>13</b> Multiple sign analysis by concatenating DO, CO, GO, KEGG, and Reactome</a></li>
<li class="chapter" data-level="14" data-path="appendix-a-comparing-performances-of-asurat-and-existing-methods.html"><a href="appendix-a-comparing-performances-of-asurat-and-existing-methods.html"><i class="fa fa-check"></i><b>14</b> Appendix A: comparing performances of ASURAT and existing methods</a><ul>
<li class="chapter" data-level="14.1" data-path="appendix-a-comparing-performances-of-asurat-and-existing-methods.html"><a href="appendix-a-comparing-performances-of-asurat-and-existing-methods.html#installation-1"><i class="fa fa-check"></i><b>14.1</b> Installation</a></li>
<li class="chapter" data-level="14.2" data-path="appendix-a-comparing-performances-of-asurat-and-existing-methods.html"><a href="appendix-a-comparing-performances-of-asurat-and-existing-methods.html#removing-the-outlier-cells-detected-by-asurat"><i class="fa fa-check"></i><b>14.2</b> Removing the outlier cells detected by ASURAT</a></li>
<li class="chapter" data-level="14.3" data-path="appendix-a-comparing-performances-of-asurat-and-existing-methods.html"><a href="appendix-a-comparing-performances-of-asurat-and-existing-methods.html#assessing-clustering-performances"><i class="fa fa-check"></i><b>14.3</b> Assessing clustering performances</a><ul>
<li class="chapter" data-level="14.3.1" data-path="appendix-a-comparing-performances-of-asurat-and-existing-methods.html"><a href="appendix-a-comparing-performances-of-asurat-and-existing-methods.html#seurat"><i class="fa fa-check"></i><b>14.3.1</b> Seurat</a></li>
<li class="chapter" data-level="14.3.2" data-path="appendix-a-comparing-performances-of-asurat-and-existing-methods.html"><a href="appendix-a-comparing-performances-of-asurat-and-existing-methods.html#monocle-3"><i class="fa fa-check"></i><b>14.3.2</b> Monocle 3</a></li>
<li class="chapter" data-level="14.3.3" data-path="appendix-a-comparing-performances-of-asurat-and-existing-methods.html"><a href="appendix-a-comparing-performances-of-asurat-and-existing-methods.html#sc3"><i class="fa fa-check"></i><b>14.3.3</b> SC3</a></li>
<li class="chapter" data-level="14.3.4" data-path="appendix-a-comparing-performances-of-asurat-and-existing-methods.html"><a href="appendix-a-comparing-performances-of-asurat-and-existing-methods.html#principal-component-analysis-pca"><i class="fa fa-check"></i><b>14.3.4</b> Principal component analysis (PCA)</a></li>
<li class="chapter" data-level="14.3.5" data-path="appendix-a-comparing-performances-of-asurat-and-existing-methods.html"><a href="appendix-a-comparing-performances-of-asurat-and-existing-methods.html#asurat"><i class="fa fa-check"></i><b>14.3.5</b> ASURAT</a></li>
<li class="chapter" data-level="14.3.6" data-path="appendix-a-comparing-performances-of-asurat-and-existing-methods.html"><a href="appendix-a-comparing-performances-of-asurat-and-existing-methods.html#comparing-clustering-validity-indices"><i class="fa fa-check"></i><b>14.3.6</b> Comparing clustering validity indices</a></li>
</ul></li>
<li class="chapter" data-level="14.4" data-path="appendix-a-comparing-performances-of-asurat-and-existing-methods.html"><a href="appendix-a-comparing-performances-of-asurat-and-existing-methods.html#cell-type-inference-in-pbmc_4000-and-pbmc_6000"><i class="fa fa-check"></i><b>14.4</b> Cell-type inference in <code>pbmc_4000</code> and <code>pbmc_6000</code></a><ul>
<li class="chapter" data-level="14.4.1" data-path="appendix-a-comparing-performances-of-asurat-and-existing-methods.html"><a href="appendix-a-comparing-performances-of-asurat-and-existing-methods.html#seurat-1"><i class="fa fa-check"></i><b>14.4.1</b> Seurat</a></li>
<li class="chapter" data-level="14.4.2" data-path="appendix-a-comparing-performances-of-asurat-and-existing-methods.html"><a href="appendix-a-comparing-performances-of-asurat-and-existing-methods.html#monocle-3-1"><i class="fa fa-check"></i><b>14.4.2</b> Monocle 3</a></li>
<li class="chapter" data-level="14.4.3" data-path="appendix-a-comparing-performances-of-asurat-and-existing-methods.html"><a href="appendix-a-comparing-performances-of-asurat-and-existing-methods.html#sc3-1"><i class="fa fa-check"></i><b>14.4.3</b> SC3</a></li>
<li class="chapter" data-level="14.4.4" data-path="appendix-a-comparing-performances-of-asurat-and-existing-methods.html"><a href="appendix-a-comparing-performances-of-asurat-and-existing-methods.html#asurat-1"><i class="fa fa-check"></i><b>14.4.4</b> ASURAT</a></li>
</ul></li>
<li class="chapter" data-level="14.5" data-path="appendix-a-comparing-performances-of-asurat-and-existing-methods.html"><a href="appendix-a-comparing-performances-of-asurat-and-existing-methods.html#identifying-cell-types-in-pbmc_4000-and-pbmc_6000"><i class="fa fa-check"></i><b>14.5</b> Identifying cell types in <code>pbmc_4000</code> and <code>pbmc_6000</code></a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="appendix-b-automatically-tuning-asurats-parameters.html"><a href="appendix-b-automatically-tuning-asurats-parameters.html"><i class="fa fa-check"></i><b>15</b> Appendix B: automatically tuning ASURATâ€™s parameters</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Introduction to ASURAT (version 0.0.0.9000)</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="normalization" class="section level1">
<h1><span class="header-section-number">Chapter 4</span> Normalizing and centering data</h1>
<p>We utilize <code>bayNorm</code> (Tang et al., Bioinformatics, 2020) and log transformation
after the addition of pseudo counts to normalize UMI-based scRNA-seq data.
Additionally, we transform the log-normalized read counts by subtracting with
the mean expression across cells.
The resulting normalized and centered data will be used for downstream
analyses.</p>
<p>Load the data.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="normalization.html#cb1-1"></a><span class="kw">source</span>(<span class="st">&quot;R/function_general.R&quot;</span>)</span>
<span id="cb1-2"><a href="normalization.html#cb1-2"></a>day1_norm &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="dt">file =</span> <span class="st">&quot;backup/01_004_day1_norm_trim_variables.rds&quot;</span>)</span>
<span id="cb1-3"><a href="normalization.html#cb1-3"></a>day7_hypo &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="dt">file =</span> <span class="st">&quot;backup/02_004_day7_hypo_trim_variables.rds&quot;</span>)</span>
<span id="cb1-4"><a href="normalization.html#cb1-4"></a>sc68_vehi &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="dt">file =</span> <span class="st">&quot;backup/03_004_sc68_vehi_trim_variables.rds&quot;</span>)</span>
<span id="cb1-5"><a href="normalization.html#cb1-5"></a>sc68_cisp &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="dt">file =</span> <span class="st">&quot;backup/04_004_sc68_cisp_trim_variables.rds&quot;</span>)</span>
<span id="cb1-6"><a href="normalization.html#cb1-6"></a>pbmc_<span class="dv">4000</span> &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="dt">file =</span> <span class="st">&quot;backup/05_004_pbmc_4000_trim_variables.rds&quot;</span>)</span>
<span id="cb1-7"><a href="normalization.html#cb1-7"></a>pbmc_<span class="dv">6000</span> &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="dt">file =</span> <span class="st">&quot;backup/06_004_pbmc_6000_trim_variables.rds&quot;</span>)</span></code></pre></div>
<p>Check the size of the gene-by-sample matrices.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="normalization.html#cb2-1"></a><span class="kw">rbind</span>(<span class="kw">dim</span>(day1_norm[[<span class="st">&quot;data&quot;</span>]][[<span class="st">&quot;raw&quot;</span>]]), <span class="kw">dim</span>(day7_hypo[[<span class="st">&quot;data&quot;</span>]][[<span class="st">&quot;raw&quot;</span>]]),</span>
<span id="cb2-2"><a href="normalization.html#cb2-2"></a>      <span class="kw">dim</span>(sc68_vehi[[<span class="st">&quot;data&quot;</span>]][[<span class="st">&quot;raw&quot;</span>]]), <span class="kw">dim</span>(sc68_cisp[[<span class="st">&quot;data&quot;</span>]][[<span class="st">&quot;raw&quot;</span>]]),</span>
<span id="cb2-3"><a href="normalization.html#cb2-3"></a>      <span class="kw">dim</span>(pbmc_<span class="dv">4000</span>[[<span class="st">&quot;data&quot;</span>]][[<span class="st">&quot;raw&quot;</span>]]), <span class="kw">dim</span>(pbmc_<span class="dv">6000</span>[[<span class="st">&quot;data&quot;</span>]][[<span class="st">&quot;raw&quot;</span>]]))</span></code></pre></div>
<pre><code>[1,] 5272 3663
[2,] 6213 1947
[3,] 6581 3923
[4,] 6347 2285
[5,] 6658 3815
[6,] 5169 4878</code></pre>
<p>Perform <code>bayNorm()</code> to infer the true copy numbers of transcripts.
Below is the default setting of <code>bayNorm()</code>.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="normalization.html#cb4-1"></a>doBayNorm &lt;-<span class="st"> </span><span class="cf">function</span>(obj){</span>
<span id="cb4-2"><a href="normalization.html#cb4-2"></a>  obj[[<span class="st">&quot;history&quot;</span>]][[<span class="st">&quot;doBayNorm&quot;</span>]][[<span class="st">&quot;MeanBETA&quot;</span>]] &lt;-<span class="st"> </span>MeanBETA</span>
<span id="cb4-3"><a href="normalization.html#cb4-3"></a>  mat &lt;-<span class="st"> </span><span class="kw">as.matrix</span>(obj[[<span class="st">&quot;data&quot;</span>]][[<span class="st">&quot;raw&quot;</span>]])</span>
<span id="cb4-4"><a href="normalization.html#cb4-4"></a>  BETA &lt;-<span class="st"> </span><span class="kw">BetaFun</span>(<span class="dt">Data =</span> mat, <span class="dt">MeanBETA =</span> <span class="fl">0.06</span>)  <span class="co"># 0.06 is the default value</span></span>
<span id="cb4-5"><a href="normalization.html#cb4-5"></a>  bay_out &lt;-<span class="st"> </span><span class="kw">bayNorm</span>(mat, <span class="dt">BETA_vec =</span> BETA[[<span class="st">&quot;BETA&quot;</span>]], <span class="dt">mode_version =</span> <span class="ot">TRUE</span>)</span>
<span id="cb4-6"><a href="normalization.html#cb4-6"></a>  obj[[<span class="st">&quot;data&quot;</span>]][[<span class="st">&quot;bayNorm&quot;</span>]] &lt;-<span class="st"> </span>bay_out</span>
<span id="cb4-7"><a href="normalization.html#cb4-7"></a>  <span class="kw">return</span>(obj)</span>
<span id="cb4-8"><a href="normalization.html#cb4-8"></a>}</span>
<span id="cb4-9"><a href="normalization.html#cb4-9"></a></span>
<span id="cb4-10"><a href="normalization.html#cb4-10"></a>day1_norm &lt;-<span class="st"> </span><span class="kw">doBayNorm</span>(<span class="dt">obj =</span> day1_norm)</span>
<span id="cb4-11"><a href="normalization.html#cb4-11"></a>day7_hypo &lt;-<span class="st"> </span><span class="kw">doBayNorm</span>(<span class="dt">obj =</span> day7_hypo)</span>
<span id="cb4-12"><a href="normalization.html#cb4-12"></a>sc68_vehi &lt;-<span class="st"> </span><span class="kw">doBayNorm</span>(<span class="dt">obj =</span> sc68_vehi)</span>
<span id="cb4-13"><a href="normalization.html#cb4-13"></a>sc68_cisp &lt;-<span class="st"> </span><span class="kw">doBayNorm</span>(<span class="dt">obj =</span> sc68_cisp)</span>
<span id="cb4-14"><a href="normalization.html#cb4-14"></a>pbmc_<span class="dv">4000</span> &lt;-<span class="st"> </span><span class="kw">doBayNorm</span>(<span class="dt">obj =</span> pbmc_<span class="dv">4000</span>)</span>
<span id="cb4-15"><a href="normalization.html#cb4-15"></a>pbmc_<span class="dv">6000</span> &lt;-<span class="st"> </span><span class="kw">doBayNorm</span>(<span class="dt">obj =</span> pbmc_<span class="dv">6000</span>)</span></code></pre></div>
<p>The following function <code>log1p_data()</code> performs log (natural logarithm)
transform of the input data with a pseudo count <code>eps</code>.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="normalization.html#cb5-1"></a>log1p_data &lt;-<span class="st"> </span><span class="cf">function</span>(obj, eps){</span>
<span id="cb5-2"><a href="normalization.html#cb5-2"></a>  obj[[<span class="st">&quot;history&quot;</span>]][[<span class="st">&quot;log1p_data&quot;</span>]][[<span class="st">&quot;eps&quot;</span>]] &lt;-<span class="st"> </span>eps</span>
<span id="cb5-3"><a href="normalization.html#cb5-3"></a>  mat &lt;-<span class="st"> </span><span class="kw">as.matrix</span>(obj[[<span class="st">&quot;data&quot;</span>]][[<span class="st">&quot;bayNorm&quot;</span>]][[<span class="st">&quot;Bay_out&quot;</span>]])</span>
<span id="cb5-4"><a href="normalization.html#cb5-4"></a>  lmat &lt;-<span class="st"> </span><span class="kw">log</span>(mat <span class="op">+</span><span class="st"> </span>eps)</span>
<span id="cb5-5"><a href="normalization.html#cb5-5"></a>  obj[[<span class="st">&quot;data&quot;</span>]][[<span class="st">&quot;log1p&quot;</span>]] &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(lmat)</span>
<span id="cb5-6"><a href="normalization.html#cb5-6"></a>  <span class="kw">return</span>(obj)</span>
<span id="cb5-7"><a href="normalization.html#cb5-7"></a>}</span>
<span id="cb5-8"><a href="normalization.html#cb5-8"></a></span>
<span id="cb5-9"><a href="normalization.html#cb5-9"></a>day1_norm &lt;-<span class="st"> </span><span class="kw">log1p_data</span>(<span class="dt">obj =</span> day1_norm, <span class="dt">eps =</span> <span class="dv">1</span>)</span>
<span id="cb5-10"><a href="normalization.html#cb5-10"></a>day7_hypo &lt;-<span class="st"> </span><span class="kw">log1p_data</span>(<span class="dt">obj =</span> day7_hypo, <span class="dt">eps =</span> <span class="dv">1</span>)</span>
<span id="cb5-11"><a href="normalization.html#cb5-11"></a>sc68_vehi &lt;-<span class="st"> </span><span class="kw">log1p_data</span>(<span class="dt">obj =</span> sc68_vehi, <span class="dt">eps =</span> <span class="dv">1</span>)</span>
<span id="cb5-12"><a href="normalization.html#cb5-12"></a>sc68_cisp &lt;-<span class="st"> </span><span class="kw">log1p_data</span>(<span class="dt">obj =</span> sc68_cisp, <span class="dt">eps =</span> <span class="dv">1</span>)</span>
<span id="cb5-13"><a href="normalization.html#cb5-13"></a>pbmc_<span class="dv">4000</span> &lt;-<span class="st"> </span><span class="kw">log1p_data</span>(<span class="dt">obj =</span> pbmc_<span class="dv">4000</span>, <span class="dt">eps =</span> <span class="dv">1</span>)</span>
<span id="cb5-14"><a href="normalization.html#cb5-14"></a>pbmc_<span class="dv">6000</span> &lt;-<span class="st"> </span><span class="kw">log1p_data</span>(<span class="dt">obj =</span> pbmc_<span class="dv">6000</span>, <span class="dt">eps =</span> <span class="dv">1</span>)</span></code></pre></div>
<p>The following function <code>centralize_data()</code> centralizes the input data on a
gene-by-gene basis.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="normalization.html#cb6-1"></a>centralize_data &lt;-<span class="st"> </span><span class="cf">function</span>(obj){</span>
<span id="cb6-2"><a href="normalization.html#cb6-2"></a>  mat &lt;-<span class="st"> </span><span class="kw">as.matrix</span>(obj[[<span class="st">&quot;data&quot;</span>]][[<span class="st">&quot;log1p&quot;</span>]])</span>
<span id="cb6-3"><a href="normalization.html#cb6-3"></a>  cmat &lt;-<span class="st"> </span><span class="kw">sweep</span>(mat, <span class="dv">1</span>, <span class="kw">apply</span>(mat, <span class="dv">1</span>, mean), <span class="dt">FUN =</span> <span class="st">&quot;-&quot;</span>)</span>
<span id="cb6-4"><a href="normalization.html#cb6-4"></a>  obj[[<span class="st">&quot;data&quot;</span>]][[<span class="st">&quot;centered&quot;</span>]] &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(cmat)</span>
<span id="cb6-5"><a href="normalization.html#cb6-5"></a>  <span class="kw">return</span>(obj)</span>
<span id="cb6-6"><a href="normalization.html#cb6-6"></a>}</span>
<span id="cb6-7"><a href="normalization.html#cb6-7"></a></span>
<span id="cb6-8"><a href="normalization.html#cb6-8"></a>day1_norm &lt;-<span class="st"> </span><span class="kw">centralize_data</span>(<span class="dt">obj =</span> day1_norm)</span>
<span id="cb6-9"><a href="normalization.html#cb6-9"></a>day7_hypo &lt;-<span class="st"> </span><span class="kw">centralize_data</span>(<span class="dt">obj =</span> day7_hypo)</span>
<span id="cb6-10"><a href="normalization.html#cb6-10"></a>sc68_vehi &lt;-<span class="st"> </span><span class="kw">centralize_data</span>(<span class="dt">obj =</span> sc68_vehi)</span>
<span id="cb6-11"><a href="normalization.html#cb6-11"></a>sc68_cisp &lt;-<span class="st"> </span><span class="kw">centralize_data</span>(<span class="dt">obj =</span> sc68_cisp)</span>
<span id="cb6-12"><a href="normalization.html#cb6-12"></a>pbmc_<span class="dv">4000</span> &lt;-<span class="st"> </span><span class="kw">centralize_data</span>(<span class="dt">obj =</span> pbmc_<span class="dv">4000</span>)</span>
<span id="cb6-13"><a href="normalization.html#cb6-13"></a>pbmc_<span class="dv">6000</span> &lt;-<span class="st"> </span><span class="kw">centralize_data</span>(<span class="dt">obj =</span> pbmc_<span class="dv">6000</span>)</span></code></pre></div>
<p>Save the objects.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="normalization.html#cb7-1"></a><span class="kw">saveRDS</span>(day1_norm, <span class="dt">file =</span> <span class="st">&quot;backup/01_005_day1_norm_normalized.rds&quot;</span>)</span>
<span id="cb7-2"><a href="normalization.html#cb7-2"></a><span class="kw">saveRDS</span>(day7_hypo, <span class="dt">file =</span> <span class="st">&quot;backup/02_005_day7_hypo_normalized.rds&quot;</span>)</span>
<span id="cb7-3"><a href="normalization.html#cb7-3"></a><span class="kw">saveRDS</span>(sc68_vehi, <span class="dt">file =</span> <span class="st">&quot;backup/03_005_sc68_vehi_normalized.rds&quot;</span>)</span>
<span id="cb7-4"><a href="normalization.html#cb7-4"></a><span class="kw">saveRDS</span>(sc68_cisp, <span class="dt">file =</span> <span class="st">&quot;backup/04_005_sc68_cisp_normalized.rds&quot;</span>)</span>
<span id="cb7-5"><a href="normalization.html#cb7-5"></a><span class="kw">saveRDS</span>(pbmc_<span class="dv">4000</span>, <span class="dt">file =</span> <span class="st">&quot;backup/05_005_pbmc_4000_normalized.rds&quot;</span>)</span>
<span id="cb7-6"><a href="normalization.html#cb7-6"></a><span class="kw">saveRDS</span>(pbmc_<span class="dv">6000</span>, <span class="dt">file =</span> <span class="st">&quot;backup/06_005_pbmc_6000_normalized.rds&quot;</span>)</span></code></pre></div>

</div>
            </section>

          </div>
        </div>
      </div>
<a href="dataqc.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="computcorr.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["Introduction to ASURAT (ver. 0.0.0.9000).pdf", "Introduction to ASURAT (ver. 0.0.0.9000).epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
